src/core/src/energy/functions.rs

```diff
@@ -364,6 +364,50 @@ impl PairPotential for BornMayerHuggins {
     }
 }
 
+/// Morse potential.
+///
+/// The following potential expression is used: `V(r) = De * (1 - exp(-a (r - re))^2 ; where 'r' is the distance between the atoms, 
+/// 're' is the equilibrium bond distance, 'De' is the well depth, and 'a' controls the 'width' of the potential
+///
+/// # Examples
+///
+/// ```
+/// use lumol::energy::Potential;
+/// use lumol::energy::MorsePotential;
+///
+/// let potential = BornMayerHuggins{a: 2.0, r0: 3.0, De: 4.0};
+/// assert_eq!(MorsePotential.energy(1.0), 3.611618461763754);
+/// assert_eq!(MorsePotential.force(2.5), 6.25);
+/// ```
+#[derive(Clone, Copy)]
+pub struct MorsePotential {
+    /// Width value
+    pub width: f64, 
+    /// Equilibrium bond distance
+    pub r0: f64,
+    /// Well depth value
+    pub depth: f64,
+}
```

Comment:
In the user doc ? I hesitated to avoid losing with the Wikipedia formula: I had written the correspondence like this "` depth` ($ From $). But I'll change that!